Frameworks/CoreText/CTLine.mm

```diff
@@ -283,16 +283,44 @@ CGRect CTLineGetImageBounds(CTLineRef line, CGContextRef context) {
 }
 
 /**
- @Status Caveat
- @Notes Values only reflect the typographical maximums for the font - string is only evaulated
-        for width
+ @Status Interoperable
+ @Notes
 */
 double CTLineGetTypographicBounds(CTLineRef lineRef, CGFloat* ascent, CGFloat* descent, CGFloat* leading) {
-    if (!lineRef) {
+    _CTLine* line = static_cast<_CTLine*>(lineRef);
+
+    if (!line || [line->_runs count] == 0) {
         return 0;
     }
 
-    _CTLine* line = static_cast<_CTLine*>(lineRef);
+    // Created with impossible values FLT_MIN which signify they need to be populated
```

Comment:
I'll add a comment. I thought FLT_MIN was along the lines of INT_MIN and the most negative value, not the smallest absolute value. I'll change it to FLT_MAX (which should be so far beyond a reasonable value that it can be considered impossible)